### ISSUE TITLE
Proform Studio bike

### ISSUE THREAD

**Issue by cagnulein** [cagnulein/qdomyos-zwift#453]
[btsnoop_hci.log](https://github.com/cagnulein/qdomyos-zwift/files/7344236/btsnoop_hci.log)

**Comment by cagnulein** [comment]
[Uploading 2021_10_13_08_10_Manual_Workout.csv…]()

**Comment by cagnulein** [comment]
[btsnoop_hci.log](https://github.com/cagnulein/qdomyos-zwift/files/7386610/btsnoop_hci.log)
[2021_10_20_08_10_Arenal_Observatory_Lodge_Interval_Ride,_La_Fortuna,_Costa_Rica.csv](https://github.com/cagnulein/qdomyos-zwift/files/7386611/2021_10_20_08_10_Arenal_Observatory_Lodge_Interval_Ride._La_Fortuna._Costa_Rica.csv)

**Comment by cagnulein** [comment]
[debug-Mon_Nov_1_06_43_11_2021.log](https://github.com/cagnulein/qdomyos-zwift/files/7458602/debug-Mon_Nov_1_06_43_11_2021.log)
<pre>More test results, and the inclination values are still off…

First all the negative incline values from the bike (from -0.5 to -15 full range) did not change the tile value, it stayed at 0

For all positive incline values (range +0.5 to +15.0), some had accurate tile values (example, 0.5 bike setting showed 0.5 incline on tile) except these:

1.5, 2.0, 4.0, 4.5, 5.0, 6.5, 7.0, 7.5, 9.0, 9.5, 10.0, 12.0, 12.5, 14.5, 15.0  - these bike settings did not change tile value (existing tile value remained) 

2.5 registered -0.1 on the tile (the only one that had a “weird” value)</pre>

**Comment by cagnulein** [comment]
[logs (3).zip](https://github.com/cagnulein/qdomyos-zwift/files/7476419/logs.3.zip)
[2021_11_04_08_11_Manual_Workout.csv](https://github.com/cagnulein/qdomyos-zwift/files/7476422/2021_11_04_08_11_Manual_Workout.csv)

**Comment by cagnulein** [comment]
[2021_11_04_17_11_Manual_Workout.csv](https://github.com/cagnulein/qdomyos-zwift/files/7481185/2021_11_04_17_11_Manual_Workout.csv)
[logs (4).zip](https://github.com/cagnulein/qdomyos-zwift/files/7481186/logs.4.zip)

**Comment by cagnulein** [comment]
[debug-Sat_Nov_6_20_07_30_2021.log](https://github.com/cagnulein/qdomyos-zwift/files/7494860/debug-Sat_Nov_6_20_07_30_2021.log)

**Comment by cagnulein** [comment]
[debug-Wed_Nov_10_14_29_32_2021.log](https://github.com/cagnulein/qdomyos-zwift/files/7519293/debug-Wed_Nov_10_14_29_32_2021.log)

Hi, it worked for the most part.

It could go up and down but only went to max 16.  After pressing + on the tile after that, it didn’t go any higher.  When I manual pressed + on the bike it went to max 26. Then when I pressed - on the tile it jumped to 16. Must be a variable thing.

I uploaded the debug log if it helps. Thanks for this, I feel we are getting close!

**Comment by cagnulein** [comment]
[debug-Wed_Nov_17_22_59_52_2021.log](https://github.com/cagnulein/qdomyos-zwift/files/7561043/debug-Wed_Nov_17_22_59_52_2021.log)
incline doesn't change on zwit

**Comment by cagnulein** [comment]
<pre>I think I see the issue but you will know better. Because i didn’t touch resistance it was probably the slowest ride I’ve done. Lol. At least I set the start resistance to be 11 so at least the avatar moved. What I noticed is that when the incline changed, the resistance would adjust up or down depending on the change in incline. So the start incline was like 3% so when the the incline went to 2%, the resistance changed down, but with the incline went up, the resistance would go up also.  The incline tile never changed. So my guess is that the app is setting resistance change, rather than incline when the Zwifit incline changes. Hopefully the debug log will show that and it’s easy to fix.

Here is the strava activity https://strava.app.link/PBEn9yWHilb</pre>
[debug-Thu_Nov_18_08_41_12_2021.log](https://github.com/cagnulein/qdomyos-zwift/files/7568439/debug-Thu_Nov_18_08_41_12_2021.log)
